src/main/java/io/r2dbc/postgresql/PostgresqlConnectionFactory.java

```diff
@@ -94,11 +94,7 @@ private static SocketAddress createSocketAddress(PostgresqlConnectionConfigurati
             return InetSocketAddress.createUnresolved(configuration.getRequiredHost(), configuration.getPort());
         }
 
-        if (configuration.isUseSocket()) {
-            return new DomainSocketAddress(configuration.getRequiredSocket());
-        }
-
-        throw new IllegalArgumentException("Cannot create SocketAddress for " + configuration);
+        return new DomainSocketAddress(configuration.getRequiredSocket());
```

Comment:
When already working on this, we should make this method more robust against absence of `DomainSocketAddress` (that is when someone excludes the epoll transport or native transport in general). Please move the instance creation into an inner static class, something along the lines of:

``` 
    static class Util {

        private static SocketAddress getDomainSocketAddress(PostgresqlConnectionConfiguration configuration) {
            return new DomainSocketAddress(configuration.getRequiredSocket());
        }

    }
```

Reply:
Ok, noted.

Reply:
@mp911de please, check it again.